packages/block-library/src/image/edit.native.js

```diff
@@ -448,7 +448,9 @@ export class ImageEdit extends Component {
 			imageDefaultSize,
 		] );
 
-		const isFeaturedImage = featuredImageId === attributes.id;
+		// Disabled on iOS until https://github.com/WordPress/gutenberg/pull/31345 is merged in to provide iOS bridge support.
+		const isFeaturedImage =
```

Comment:
Following [this comment](https://github.com/WordPress/gutenberg/pull/31681/#issuecomment-838218557), I found that the bug with the page template still exists against [this open PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/16427), I think a check in place for whether the `featuredImageId` is `undefined` may be what's needed to eliminate the bug altogether:  
```suggestion
		const isFeaturedImage = featuredImageId
			? featuredImageId === attributes.id
			: false;
```

Reply:
+1 for a fix based on checking for undefined, that would also cover the same case on Android.

While at it, I think we should check for undefined directly instead of the plain `featuredImageId ? ...` in the suggestion Siobhan. If featuredImageId is `0` it will check as falsy but it might be a valid image id. So, wdyt about using:
`const isFeaturedImage = (typeof featuredImageId !== 'undefined') && (featuredImageId === attributes.id)`?

Reply:
Thanks for the thoughts here! I will go ahead and make this change. I thought the default feature image id would also be 0 but since it can be `undefined` this approach will cover all the cases.

Reply:
Implemented here 9583973